include/envoy/api/os_sys_calls.h

```diff
@@ -170,6 +170,11 @@ class OsSysCalls {
    * @see man 2 accept. The fds returned are configured to be non-blocking.
    */
   virtual SysCallSocketResult accept(os_fd_t socket, sockaddr* addr, socklen_t* addrlen) PURE;
+
+  /**
+   * @see man TCP_INFO. Get the tcp info for the socket.
+   */
+  virtual SysCallBoolResult socketTcpInfo(os_fd_t sockfd, tcp_info* tcp_info) PURE;
```

Comment:
In the current use case, we just need RTT value. I think at this point it would make more sense to define EnvoyTcpInfo with just `RTT` member. It was suggested in the original PR but I think it makes a lot more sense now. In this case you do not need to pull in tcp_info struct into Envoy's headers.

Reply:
I am planning to add the header in this header? Is there a better place for small structs?

Reply:
Fixed in 7c9137b7a3fb86ccae63cbd4018683cc47b7cb4d.

Feel free to open the conversation again if you think that the struct should be moved into a different header